pyqtgraph/graphicsItems/PlotItem/PlotItem.py

```diff
@@ -284,65 +285,155 @@ def method(self, *args, **kwargs):
                 return getattr(self.vb, name)(*args, **kwargs)
             method.__name__ = name
             return method
-        
+
         locals()[m] = _create_method(m)
-        
+
     del _create_method
-    
-    def setAxisItems(self, axisItems=None):
+
+    def removeAxis(
+        self,
+        name: str,
+        unlink: bool = True,
+
+    ) -> Optional[AxisItem]:
         """
-        Place axis items as given by `axisItems`. Initializes non-existing axis items.
-        
+        Remove an axis from the contained axis items
+        by ```name: str```.
+
+        This means the axis graphics object will be removed
+        from the ``.layout: QGraphicsGridLayout`` as well as unlinked
+        from the underlying associated ``ViewBox``.
+
+        If the ``unlink: bool`` is set to ``False`` then the axis will
+        stay linked to its view and will only be removed from the
+        layoutonly be removed from the layout.
+
+        If no axis with ``name: str`` is found then this is a noop.
+
+        Return the axis instance that was removed.
+
+        """
+        entry = self.axes.pop(name, None)
+
+        if not entry:
+            return
+
+        axis = entry['item']
+        self.layout.removeItem(axis)
+        axis.scene().removeItem(axis)
+        if unlink:
+            axis.unlinkFromView()
+
+        self.update()
+
+        return axis
+
+    # NOTE: seriously.. y'all
+    # Why do we need to always have all axes created? I don't understand
```

Comment:
This is a pretty serious design thing I noticed.
I don't grok at all why all axes must always be created **other then** that when you don't create them the `'bottom'` axis geometry seems to still be used to in the view calculations resulting in `.viewRect()` style calcs to compute wrong?!